mu4/domain/notation/internal/msczmetareader.cpp

```diff
@@ -0,0 +1,345 @@
+        return meta;
+    }
+
+    QByteArray rootBuffer = zipReader.fileData(rootfile);
+    if (rootBuffer.isEmpty()) {
+        auto fil = zipReader.fileInfoList();
+        for (const MQZipReader::FileInfo& fi : fil) {
+            if (mu::strings::endsWith(fi.filePath.toStdString(), ".mscx")) {
+                rootBuffer = zipReader.fileData(fi.filePath);
+                break;
+            }
+        }
+    }
+
+    QXmlStreamReader xmlReader(rootBuffer);
+    meta = doReadMeta(xmlReader);
+
+    meta.val.thumbnail = loadThumbnail(&zipReader);
+
+    return meta;
+}
+
+void MsczMetaReader::doReadBox(QXmlStreamReader& xmlReader, RawMeta& meta) const
+{
+    while (xmlReader.readNextStartElement()) {
+        if (xmlReader.name() == "Text") {
+            bool isTitle = false;
+            bool isComposer = false;
+            while (xmlReader.readNextStartElement()) {
+                const QStringRef& tag(xmlReader.name());
+
+                if (tag == "style") {
+                    QString val(xmlReader.readElementText().toLower());
+
+                    if (val == "title" || val == "2") {
+                        isTitle = true;
+                    } else if (val == "composer" || val == "4") {
+                        isComposer = true;
+                    } else {
+                        xmlReader.skipCurrentElement();
+                    }
+                } else if (tag == "text") {
+                    if (isTitle) {
+                        meta.titleStyle = formatFromXml(xmlReader.readElementText(
+                                                            QXmlStreamReader::IncludeChildElements));
+                    } else if (isComposer) {
+                        meta.composerStyle
+                            = formatFromXml(xmlReader.readElementText(QXmlStreamReader::IncludeChildElements));
+                    } else {
+                        xmlReader.skipCurrentElement();
+                    }
+                } else if (tag == "html-data") {
+                    if (isTitle) {
+                        meta.titleStyleHtml
+                            = formatFromXml(xmlReader.readElementText(QXmlStreamReader::IncludeChildElements));
+                    } else if (isComposer) {
+                        meta.composerStyleHtml
+                            = formatFromXml(xmlReader.readElementText(QXmlStreamReader::IncludeChildElements));
+                    } else {
+                        xmlReader.skipCurrentElement();
+                    }
+                } else {
+                    xmlReader.skipCurrentElement();
+                }
+            }
+        } else {
+            xmlReader.skipCurrentElement();
+        }
+    }
+}
+
+MsczMetaReader::RawMeta MsczMetaReader::doReadRawMeta(QXmlStreamReader& xmlReader) const
+{
+    RawMeta meta;
+
+    while (xmlReader.readNextStartElement()) {
+        const QStringRef& tag(xmlReader.name());
+        if (tag == "work-title") {
```

Comment:
Where should this tag stem from?

Reply:
This is from some old formats, maybe version 2.
MsczMetaReader I wrote some time ago for a mobile application. Tested on 2000 files of different versions, and figured out how to store meta information in different versions of files.

Reply:
No. MuseScore 2.0 already had workTitle, 1.x didn't have those tags at all

Reply:
hmm, then Elnur can remove them

Reply:
Hmm, maybe not, this (and more!) indeed is done and checked for in read114.cpp, read206.cpp and read301.cpp as well as in exportxml.cpp